.gitlab-ci.yml

```diff
@@ -237,6 +243,37 @@ gitleaks:
   script: gitleaks detect -v --source .
 
 
+## Jobs running on Slurm cluster
+
+# factored out command to download the datasets, cmake, and build in non-debug mode
+.setup_and_build_ndebug: &setup_and_build_ndebug
+  - ./tools/downloadDatasets.sh <<< "yes"
+  - mkdir -p install build && cd ./build
+  - ../bootstrap.sh --prefix=../install --with-datasets=../datasets
+  - make -j$(nproc)
+
+tests_performance_slurm:
+  rules:
+    - if: $SLURM_TESTS_ENABLED == "yes"
+  tags:
+    - slurm
+  variables:
+    CI_SLURM_PARTITION: "Zen3"
+    CI_SLURM_NNODES: "1"
+    CI_SLURM_NTASKS: "1"
+    CI_SLURM_CPUS_PER_TASK: "64"
+    CI_SLURM_MEM_PER_NODE: "64G"
+    CI_SLURM_TIMELIMIT: "00-10:00:00" # 10 hours
+    SLURM_JOB_START_TIMEOUT_SECONDS: "3600" # 1 hour timeout to switch from PENDING to RUNNING
```

Comment:
These values could be exposed as global variables and dynamically changed by the user triggering from the web-page, should they ?

Reply:
If possible and easy to do it would be great indeed!